squire_lib/src/pairings/swiss_pairings.rs

```diff
@@ -138,15 +136,13 @@ impl SwissPairings {
             match_size,
             repair_tol,
         );
-        while count < max_count && !pairings.rejected.is_empty() {
-            count += 1;
+
+        for _ in 0..100 {
+            if pairings.rejected.is_empty() {
+                break;
+            }
             let grouped_plyrs: GroupMap<_, _> = plyrs_and_scores.iter().cloned().collect();
-            plyrs.extend(
-                grouped_plyrs
-                    .iter_right()
-                    .flat_map(|r| grouped_plyrs.get_left_iter(r).unwrap())
-                    .cloned(),
-            );
+            plyrs.extend(grouped_plyrs.iter_left());
```

Comment:
These two iteration methods are not equivalent. Like a normal `Hashmap`, a `GroupMap` does not guarantee an order while iterating over its elements. This is doubly so for a `GroupMap` as the above applies to both the left and right sides (internally, a `GroupMap` is actually two `HashMap`s).

The old iteration method would ensure that players that have the same scores would be next two each other in the final `Vec`. Looking at the iterator code in `cycle_map`, it looks like this would work, however:
```suggestion
            players.extend(grouped_plyers.iter().filter_map(|(plyr, _)| plyr).cloned())
```

Reply:
I do understand your confusion, though. CycleMap was my first ever crate. I made it just a few months over I learned Rust. It is definitely in need of some TLC.